.github/workflows/release.yml

```diff
@@ -51,6 +51,15 @@ jobs:
             echo "Tag should match pom.xml project.version"
             exit 1
           fi
+      - uses: ./.github/workflows/maven-goal
```

Comment:
Do we even need this here? Don't we check for exact equality with the provided `end.RELEASE_VERSION` above?

Someone would need to actually provide `1.6.0-SNAPSHOT` here to accidentally release a snapshot?

Reply:
> Do we even need this here? Don't we check for exact equality with the provided end?RELEASE_VERSION above?

You are right!

Although I'm a bit confused about the reason for using the `inputs.version`, IIUC, it should be the `pre-release` workflow, the one bumping the version and the `release` workflow, the one using the `pom.xml` file, to do the release. Hence `inputs.version` is not something we need at all, the `buildkite` pipeline is not using it.

Reply:
> Someone would need to actually provide 1.6.0-SNAPSHOT here to accidentally release a snapshot?

Or change the pom.xml version in a different and use a different `inputs.ref`.

I think the validation is still valid and in fact, the checkout for the given ref is needed, I'll do the change now